ldap/servers/slapd/back-ldbm/db-bdb/bdb_config.c

```diff
@@ -228,6 +228,103 @@ bdb_config_db_lock_set(void *arg, void *value, char *errorbuf, int phase, int ap
     return retval;
 }
 
+static void *
+bdb_config_db_lock_monitoring_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+
+    return (void *)((uintptr_t)(li->li_new_dblock_monitoring));
+}
+
+static int
+bdb_config_db_lock_monitoring_set(void *arg, void *value, char *errorbuf __attribute__((unused)), int phase __attribute__((unused)), int apply)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    int retval = LDAP_SUCCESS;
+    int val = (int)((uintptr_t)value);
+
+    if (apply) {
+        if (CONFIG_PHASE_RUNNING == phase) {
+            li->li_new_dblock_monitoring = val;
+            slapi_log_err(SLAPI_LOG_NOTICE, "bdb_config_db_lock_monitoring_set",
+                          "New nsslapd-db-lock-monitoring value will not take affect until the server is restarted\n");
+        } else {
+            li->li_new_dblock_monitoring = val;
+            li->li_dblock_monitoring = val;
+        }
+    }
+
+    return retval;
+}
+
+static void *
+bdb_config_db_lock_pause_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+
+    return (void *)((uintptr_t)(slapi_atomic_load_32(&(li->li_dblock_monitoring_pause), __ATOMIC_RELAXED)));
+}
+
+#define DBLOCK_MONITORING_PAUSE_DEFAULT 100
```

Comment:
Just a comment saying that 100 = 1sec

Reply:
Hm, but I thought **PR_MillisecondsToInterval** accepts milliseconds (0.001 of a second) and converts it to Interval Time.
Am I confused? And do I miss something obvious?..
https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Reference/NSPR_functions#interval_time

Reply:
Sorry I was first wrong (assuming 1sec) and also not clear. if '100' stands for 100ms then adding a comment helps to know what this value is for: /* by default pausing 100ms between each db_stat */.

Is 100ms not to intensive to gather statistics by default ?

Reply:
I agree, I've changed the default to 1000ms.